quickwit/quickwit-opentelemetry/src/otlp/logs.rs

```diff
@@ -343,25 +341,21 @@ impl OtlpGrpcLogsService {
                 }
             }
         }
+        let mut doc_batch = DocBatchBuilder::new(OTEL_LOGS_INDEX_ID.to_string()).json_writer();
         for log_record in log_records {
-            let current_doc_batch_len = doc_batch.concat_docs.len();
-            if let Err(error) = serde_json::to_writer(&mut doc_batch.concat_docs, &log_record.0) {
-                error!(error=?error, "Failed to JSON serialize log record.");
-                error_message = format!("Failed to JSON serialize log record: {error:?}");
+            if let Err(error) = doc_batch.ingest_doc(&log_record.0) {
+                error!(error=?error, "Failed to JSON serialize span.");
+                error_message = format!("Failed to JSON serialize span: {error:?}");
                 num_parse_errors += 1;
-                continue;
             }
-            let new_doc_batch_len = doc_batch.concat_docs.len();
-            let span_json_len = new_doc_batch_len - current_doc_batch_len;
-            doc_batch.doc_lens.push(span_json_len as u64);
         }
         let current_span = RuntimeSpan::current();
         current_span.record("num_log_records", num_log_records);
-        current_span.record("num_bytes", doc_batch.concat_docs.len());
+        current_span.record("num_bytes", doc_batch.len());
         current_span.record("num_parse_errors", num_parse_errors);
 
         let parsed_spans = ParsedLogRecords {
-            doc_batch,
+            doc_batch: doc_batch.build(),
```

Comment:
It's unclear to me whether this change causes double-counting in metrics, or fix not counting at all

Reply:
I don't think we are double counting since I removed increment in add_doc and moved it into build() and I cannot find any other places where it would be incremented, but it seems we were not counting at all in the past. @guilload is there a reason not to count these records in logs.rs and trace.rs?

Reply:
I was not aware that counting was performed when building the doc batch. `IngestApiService::ingest` might be a better place.

Reply:
Thanks! I will take a look at pushing it back there then. It is a bit weird since we are counting bytes and records in different places, but I wasn't sure if there was a reason for it.